frontend/src/components/ProjectTeacherComponent.tsx

```diff
@@ -1,193 +1,187 @@
-                                    type="number"
-                                    value={max_students}
-                                    onChange={e => setMaxStudents(parseInt(e.target.value))}
-                                />
-                            </div>
-                        </div>
-                    }
-                </div>
-            </div>
-
-        </div>
-    )
-        ;
-}
+import {JSX, useState} from "react";
+import Inputfield from "./Inputfield.tsx";
+import {SelectionBox} from "./SelectionBox.tsx";
+import 'react-calendar/dist/Calendar.css';
+import {FaUpload} from "react-icons/fa";
+import {ProjectTeacher, Value} from "../types/project.ts";
+import "../assets/styles/teacher_components.css"
+import SimpleTests from "./SimpleTests/SimpleTests.tsx";
+import {TeacherOrStudent} from "./SimpleTests/TeacherOrStudentEnum.tsx";
+import Calendar from "react-calendar";
+import {useTranslation} from 'react-i18next';
+import Switch from "react-switch";
+
+export function ProjectTeacherComponent(props: { project: ProjectTeacher }): JSX.Element {
+
+    const {t} = useTranslation();
+
+    const [projectName, setProjectName] = useState<string>(props.project.projectName)
+    const [courseName, setCourseName] = useState<string>(props.project.courseName)
+    const [hours, setHours] = useState<number>(props.project.hours);
+    const [minutes, setMinutes] = useState<number>(props.project.minutes);
+    const [deadline, setDeadline] = useState<Value>(props.project.deadline);
+    const [description, setDescription] = useState(props.project.description);
+    const [max_students, setMaxStudents] = useState(props.project.maxGroupMembers);
+    const [requiredFiles, setRequiredFiles] = useState(props.project.requiredFiles);
+    // Deze wordt niet gebruikt. Dit zit verwerkt in het json-object als OnlyPresentConstraint.
+    // const [otherFilesAllow, setOtherFilesAllow] = useState(props.project.otherFilesAllow);
+    const [groupProject, setGroupProject] = useState(props.project.groupProject);
+
+    // helpers
+    const [showCalender, setCalender] = useState(props.project.deadline !== null);
+    const [showGroup, setGroup] = useState(props.project.groupProject);
+
+    const expandDeadline = () => {
+        setCalender(!showCalender);
+    };
+
+    const expandGroup = (checked: boolean) => {
+        setGroup(!showGroup);
+        setGroupProject(checked);
+    };
+
+    const course_options = props.project.all_courses.map(course => course.course_name);
+
+    const hours_array = Array.from({length: 24}, (_, index) => index.toString().padStart(2, '0'));
+    const minutes_array = Array.from({length: 60}, (_, index) => index.toString().padStart(2, '0'));
+
+    // SimpleTests
+    const [requiredFilesHasChanged, setRequiredFilesHasChanged] = useState(false);
+    if (requiredFilesHasChanged) {if (!requiredFilesHasChanged) {console.log("")}} // eslint prevent error

```

Comment:
Waarover klaagt eslint mss?

Reply:
De inkomende json string wordt geparsed naar een any object, maar typescript heeft dat niet graag.

Reply:
In ts kan je type casting doen met het `as` keyword.  Heb jij dit al geprobeerd?

Reply:
Ja, dat heb ik soms wel kunnen gebruiken, maar niet overal. Er zijn twee problemen daarbij. Het eerste is dat ik de verschillende soorten constraints omzet naar mijn eigen soort constraint met slechts 1 parameter value in plaats van verschillende soorten zoals file_name, file_or_directory_name, zip_name etc. en wat extra info voor de frontend zoals zichtbaarheid, diepte etc. Het tweede probleem is dat ik de recursieve structuur omzet naar een gewone lijst om gemakkelijk te kunnen mappen. Ik kan dus niet zomaar as oproepen op de ganse json. Uiteindelijk komt het er op neer dat het gewoon te veel werk zou zijn om in dit geval aan de typescript voorwaarden te voldoen.